gcp-native-ts-k8s-ruby-on-rails-postgresql/infra/cluster.ts

```diff
@@ -0,0 +1,96 @@
+// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.
+
+import * as gcp from "@pulumi/gcp-native";
+import * as k8s from "@pulumi/kubernetes";
+import * as pulumi from "@pulumi/pulumi";
+import { clusterNodeCount, clusterNodeMachineType, clusterPassword, clusterUsername, masterVersion, project, region, zone } from "./config";
+
+const clusterName = "gke-native";
+const cluster = new gcp.container.v1.Cluster("cluster", {
+    projectsId: project,
+    locationsId: region,
+    clustersId: clusterName,
+    parent: `projects/${project}/locations/${region}`,
+    initialClusterVersion: "1.18.16-gke.2100",
+    initialNodeCount: 1,
+    masterAuth: { username: clusterUsername, password: clusterPassword },
+    name: clusterName,
+    network: `projects/${project}/global/networks/default`,
+    nodeConfig: {
+        oauthScopes: [
+            "https://www.googleapis.com/auth/devstorage.read_only",
+            "https://www.googleapis.com/auth/logging.write",
+            "https://www.googleapis.com/auth/monitoring",
+            "https://www.googleapis.com/auth/service.management.readonly",
+            "https://www.googleapis.com/auth/servicecontrol",
+            "https://www.googleapis.com/auth/trace.append",
+            "https://www.googleapis.com/auth/compute",
+        ],
+    },
+});
+
+const nodepoolName = "nodepool";
+const nodePool = new gcp.container.v1.ClusterNodePool(`primary-node-pool`, {
```

Comment:
GCP question, really: is this needed? Doesn't the Cluster come with a built-in pool?

Reply:
General guidance from Google and accepted practice in the community seems to be to prefer managing nodepools explicitly: https://github.com/hashicorp/terraform-provider-google/issues/475#issuecomment-383529139https://github.com/hashicorp/terraform-provider-google/issues/475#issuecomment-383529139. In fact, the TF provider has a `remove_default_node_pool` option where the provider waits for the cluster to come up and then deletes the default nodepool explicitly. I don't believe GKE lets you create a cluster without a nodepool though and initialNodeCount must be > 0. While we could remove the nodepool creation and use the default nodepool figured in the example it would be illustrative to create a nodepool. e.g. in this case the nodepools will use preemptible VM.